rfc/20240725-exclude-resources.md

```diff
@@ -0,0 +1,117 @@
+- Running `tofu plan -exclude=null_resource.b -exclude=null_resource.c` would exclude `null_resource.b`, `null_resource.c` and also `null_resource.d` which depends on one of them (or in this case - both of them)
+- Running `tofu plan -exclude=null_resource.a -exclude=null_resource.b` would create an empty plan, since all resources depend on `null_resource.a`
+
+When destroying:
+- Running `tofu plan -destroy -exclude=null_resource.b` will result in a plan to destroy `null_resource.c` and `null_resource.d`
+
+Note that a resource is dependent on another resource not just by direct resource dependency:
+
+```hcl
+locals {
+  b = null_resource.a.id
+}
+
+resource "null_resource" "a" {
+  
+}
+
+resource "null_resource" "c" {
+  triggers = {
+    b = local.b
+  }
+}
+```
+
+In the example above, if you run `tofu plan -target=null_resource.a`, then both `null_resource.a` and `null_resource.c` will be excluded from the plan. `null_resource.c` depends on a local which in itself depends on `null_resource.a`
+
+**Note**: For now, `-exclude` and `-target` flag should not be allowed to be used in conjunction. In the future, we might allow them both to be used in conjunction, with the `-exclude`d resource taking precedence. However, this approach would require a deeper dive into it
+**Note 2**: When using the `-target` flag, on an apply from a stored plan file, the flag is completely ignored. So, the behaviour would be the same for the `-exclude` flag
+
+#### Outputs
+
+When planning with an `-exclude` flag, only outputs that rely on **at least one resource** that was not excluded should be recalculated. 
+
+This is the inverted approach to the `-target` flag, for which outputs are only recalculated if all resources that it depends all are targeted
+
+#### Data Sources
+
+Like with the `-target` flag, supplying an `-exclude` flag means that no data sources are refreshed, even if they are technically dependent on resources that are not excluded.
+
+This also means that any dependency on a data source is not considered at all when calculating whether a resource or an output is dependent on a non-excluded resource
+
+#### Cloud
+
+Unlike `-target` flag, which is passed to the cloud backend in remote runs, the `-exclude` flag will not be passed to cloud backends. This is due to a technical limitation, with the cloud client and API calls being managed by `go-tfe`.
+
+### Technical Approach
+
+The technical approach of this should be pretty simple and very similar to how targeted resources work.
+
+Mainly:
+- Add `Excludes` alongside `Targets` pretty much anywhere applicable (`Operation`, `NodeAbstractResource`)
+- Adapt `GraphNodeTargetable` to also have `SetExcludes`, for dynamic expansion
+- In the `TargetTransformer`, remove any excluded resource or resource depending on an excluded resource from the graph
+
+### Open Questions
+
+- Is `-exclude` the correct name for the flag? Maybe `-target-exclude`?
+
+### Future Considerations
+
+## Potential Alternatives
```

Comment:
Similar to the `tofu plan -light` issue (https://github.com/opentofu/opentofu/issues/1703), cli tools could potentially "simulate" this.  This alternative would be either slow or inaccurate, and thus unsuitable for this use case.

Reply:
Added mention of this in cf847ad92848b6eaeb2ceeb809c54b1cdf1aae87